tests/browser/custom-reporter.ts

```diff
@@ -0,0 +1,88 @@
+import type {
+  FullConfig,
+  FullResult,
+  Reporter,
+  Suite,
+  TestCase,
+  TestResult,
+} from '@playwright/test/reporter';
+import { writeFileSync } from 'fs';
+import path from 'path';
+
+class MyReporter implements Reporter {
+  private passCount = 0;
+  private failCount = 0;
+  private skipCount = 0;
+  private failingTests: { path: string; name: string }[] = [];
+  private skippedTests: { path: string; name: string }[] = [];
+
+  onBegin(config: FullConfig, suite: Suite) {
+    console.log(`Starting the run with ${suite.allTests().length} tests`);
+  }
+
+  onTestEnd(test: TestCase, result: TestResult) {
+    const getHierarchyPath = (test: TestCase): string => {
+      const path: string[] = [];
+      let parent: Suite | undefined = test.parent;
+      while (parent) {
+        path.unshift(parent.title || '');
+        parent = parent.parent;
```

Comment:
Maybe use optional chaining on parent.title and parent.parent to prevent this blowing up?

Reply:
@jack-ryan-nava-pbc, I think `parent.title` and `parent.parent` are safe in this context, because the `while` loop condition ensures these lines will only execute if `parent` is truthy

Reply:
I'm being over anxious and just wondering if there could be situation where `parent` exists but without the fields of title or parent attached?

Reply:
@jack-ryan-nava-pbc, Not over-anxious at all—it’s a good point! Based on the `Suite` type, `title` is always present, and `parent` is optional. The `while (parent)` condition ensures that `parent` is truthy before we access it, so the current implementation is safe. Let me know if you’d still prefer adding optional chaining for extra clarity, though I don’t think it’s strictly necessary here.

Reply:
Perfect. Thanks for confirming @tamara-corbalt!